src/core/server/saved_objects/service/saved_objects_client.ts

```diff
@@ -107,6 +107,26 @@ export interface SavedObjectsUpdateOptions extends SavedObjectsBaseOptions {
   refresh?: MutatingOperationRefreshSetting;
 }
 
+/**
+ *
+ * @public
+ */
+export interface SavedObjectsAddNamespacesOptions {
+  /** An opaque version number which changes on each successful write operation. Can be used for implementing optimistic concurrency control. */
+  version?: string;
+  /** The Elasticsearch Refresh setting for this operation */
+  refresh?: MutatingOperationRefreshSetting;
+}
+
+/**
+ *
+ * @public
+ */
+export interface SavedObjectsRemoveNamespacesOptions {
+  /** The Elasticsearch Refresh setting for this operation */
```

Comment:
`removeNamespaces` is analogous to `delete`, which doesn't allow for specifying a version. IMO the optimistic concurrency control is only there to prevent conflicts (e.g., document attribute `foo` is currently equal to `bar`, and I want to set it to `baz`... if the version changed, don't execute the update because `foo` might have changed in the meantime). A "delete" or "remove" operation doesn't seem to have the same opportunity for conflict/error.

Reply:
However, since `removeNamespaces` is a two-step operation, I think we should use the version of the document that was obtained from the `get`.

What do you think?

Reply:
> However, since `removeNamespaces` is a two-step operation, I think we should use the version of the document that was obtained from the `get`.
> 
> What do you think?

I agree! That was supposed to be my followup comment, but then I got distracted and